### DISCUSSION TITLE
Nested colspanx and rowspanx don't work

### DISCUSSION THREAD

**Discussion by mangkoran** [PgBiel/typst-tablex#104] [Q&A] [answered]
First of all, thank you for your work (and other contributors) on this library. I just started to learn Typst recently, and currently I am in the middle of writing my thesis using Typst and found this library which helps me a lot.

I would like to nest colspanx and rowspanx. However, I encountered error.

![image](https://github.com/PgBiel/typst-tablex/assets/57625126/12493be2-ca1d-4099-896c-8874397ee556)

```typst
#tablex(
  columns: (2fr, 1fr, 2fr, 1fr),
  colspanx(4)(rowspanx(2)[a]),
)
```

I followed the example [here](https://github.com/PgBiel/typst-tablex#basic-types-and-functions).

**Comment by PgBiel** [comment, marked as answer]
Hello,

Within Typst's syntax, you can write either `function(a, b)`, `function(a, [b])` or `function(a)[b]` (the last one being a shorthand for the previous one). `function(a)(b)` would first call `function(a)` and then attempt to call its result as a function with the parameter `b`. Here, thus, you are calling `colspanx(4)` (which fails because it requires content) and then trying to call its result as a function with the single argument `rowspanx(2)[a]`.

Therefore, what you meant to write is `colspanx(4, rowspanx(2)[a])`. Hope this helps!

**Reply by mangkoran** [reply]
Thank you for your reply! I have tried it and it now compiles without error. I think the documentation should be updated then. Let me know if you need help for it 😉

![image](https://github.com/PgBiel/typst-tablex/assets/57625126/c1a20ee3-a2c0-418a-b7ae-ddebd649ba4c)

However, it is somewhat not what I expect.

![image](https://github.com/PgBiel/typst-tablex/assets/57625126/1b04ae72-4771-4298-bfae-47be7c1f9b79)

```typst
#tablex(
  columns: (auto, auto, auto, auto),
  colspanx(4, rowspanx(3)[ONE]),
  [TWO], [THREE], [FOUR], [FIVE],
)
```

It seems "ONE" is stacked in place, which is not what I expect.

**Reply by PgBiel** [reply]
> Thank you for your reply! I have tried it and it now compiles without error. I think the documentation should be updated then. Let me know if you need help for it 😉

Thanks for pointing it out! Seems like a mistake was made 😅

Feel free to open a PR to main suggesting a fix! Just click `README.md`, press the edit button at the top (GitHub will create a fork and a branch for you), edit the line, save and create the pull request!

> 
> However, it is somewhat not what I expect.
> 
> (snip)
>
> ```typst
> #tablex(
>   columns: (auto, auto, auto, auto),
>   colspanx(4, rowspanx(3)[ONE]),
>   [TWO], [THREE], [FOUR], [FIVE],
> )
> ```
> 
> It seems "ONE" is stacked in place, which is not what I expect.

This seems like a bug; can you open an issue for this? Thanks! 😄 

In the meantime you can manually assign sizes to each row (note that you have 4 rows) with `rows: 2em` for example (if they all have the same size) or `rows: (0pt, 0pt, 2em, auto)` or something similar.
